migration/sql-files/112-cascading-delete.sql

```diff
@@ -0,0 +1,19 @@
+-- Change foreign key from "identities" to "users" to cascade.
+ALTER TABLE identities
+    ADD FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE,
+    DROP CONSTRAINT identities_user_id_users_id_fk;
+
+-- Change foreign key from "comment_revisions" to "identities" to ON DELETE CASCADE.
+ALTER TABLE comment_revisions
+    ADD FOREIGN KEY (modifier_id) REFERENCES identities(id) ON DELETE CASCADE,
+    DROP CONSTRAINT comment_revisions_identity_fk;
+
+-- Change foreign key from "work_item_link_revisions" to "identities" to ON DELETE CASCADE.
+ALTER TABLE work_item_link_revisions
+    ADD FOREIGN KEY (modifier_id) REFERENCES identities(id) ON DELETE CASCADE,
+    DROP CONSTRAINT work_item_link_revisions_modifier_id_fk;
+
+-- Change foreign key from "work_item_revisions" to "identities" to ON DELETE CASCADE.
+ALTER TABLE work_item_revisions
```

Comment:
I'm not sure this will turn out good. Suppose Alice created a work item and then Bob Updates it. If the user Bob shall be deleted you delete the revision that updated the work item. Is that really our intention?

Reply:
@kwk Agreed this is debatable in term of functionalities. I went the easier way, updating the FK constraint to be on cascade. 

Mainly we have 2 use-cases:
- Atm we clear the spaces, wit, comments associated to a space a user owned. So deleting the revisions of a wit which is going to be deleted is ok. I think this is the primary use-case we're targeting. 
- For a space a user does not own, what should happen if we delete a user, should we keep the comment without owner name? (like an anonymous revision). We could drop the constraint the FK altogether to achieve that. 

cc @sbose78 what should we do for the second use-cases: a space a user (we want to delete) has contributed to without owning it?

Reply:
My point could be enforced if Alice created a bug and Bob marks it as closed. Even though we persist the change in the work item (here: in the bug) itself, the history is lost. Generally we should avoid losing information with the exception of GDPR relevant data.

Reply:
Good point @kwk , we should let the work item revisions remain given that we lose trace-ability of changes in an object that wasn't owned by the user , but had been modified by the user who was deleted. 

@corinnekrych , since the comment was owned by the user - it's fine to delete it.

Reply:
> @corinnekrych , since the comment was owned by the user - it's fine to delete it.

@sbose78 why delete the comment? What about replies to that comment? They will also be deleted, right?

Reply:
@kwk @sbose78 Let me remove those 2 foreign keys `work_item_link_revisions_modifier_id_fkey` and `work_item_revisions_modifier_id_fkey`.

@kwk Looking in the UI it seems the UI [is checking if creator details are available](https://github.com/fabric8-ui/fabric8-ui/blob/master/packages/planner/src/app/components_ngrx/work-item-detail/work-item-detail.component.html#L19) so it seems a wit with not found user will not make the UI crash. we should eb fine leaving those revisions, they will be displayed without owner displayed.